Calculator/src/main/java/com/microsoft/device/display/samples/calculator/models/CalculatorModel.kt

```diff
@@ -0,0 +1,170 @@
+package com.microsoft.device.display.samples.calculator.models
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import com.microsoft.device.display.samples.calculator.ui.pages.historyModel
+import kotlin.math.roundToInt
+
+class CalculatorModel() {
+    var x: String by mutableStateOf("0")
+    var y: String by mutableStateOf("0")
+    // if we are editing x or y
+    var isOnX: Boolean by (mutableStateOf(true))
+    // if we are overwriting the current variable by typing, or adding to it by typing
+    var overWrite: Boolean by (mutableStateOf(true))
+    var degrees: Boolean by (mutableStateOf(true))
+    var currentEquation: Equation by mutableStateOf(Equation.NONE)
+
+    private fun assertEquation() {
+        val prevX = this.x
+        val prevY = this.y
+        when (currentEquation) {
+            Equation.ADD -> add()
+            Equation.SUB -> sub()
+            Equation.MUL -> mul()
+            Equation.DIV -> div()
+            Equation.POW -> power()
+            Equation.MOD -> modulus()
+            else -> {}
+        }
+        decimalReformat()
+        if (currentEquation != Equation.NONE) historyModel.addToRecords(Record(prevX, prevY, currentEquation, this.x))
```

Comment:
I would separate this to two lines for better readability - I think in general the main reason to keep if statements on one line if is you're using them for assignments (ex: `val example = if (flag) "flag is true" else "flag is false"`)